### PR TITLE
Add support for animationend event

### PR DESCRIPTION
**Pull request by mrobinson**
This is triggered when an animation finishes. This is a high priority
because it allows us to start rooting nodes with animations in the
script thread.

This doesn't yet cause a lot of tests to pass because they rely on the
existence of `Document.getAnimations()` and the presence of
`animationstart` and `animationiteration` events.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [ ] These changes fix #___ (GitHub issue number if applicable)

- [x] There are tests for these changes OR
- [ ] These changes do not require tests because ___

### CODE REVIEW

File: components/script/script_thread.rs [hunk 1]
```diff
@@ -2916,66 +2926,91 @@ impl ScriptThread {
     /// Handles firing of transition-related events.
     ///
     /// TODO(mrobinson): Add support for more events.
-    fn handle_transition_event(
+    fn handle_transition_or_animation_even(
```
Comment (jdm) [inline components/script/script_thread.rs:2929]: event

**Review by jdm — APPROVED** [review]
Looks great!

**Review by emilio — APPROVED** [review]
Looks sane, thanks.

File: components/script/dom/webidls/EventHandler.webidl [hunk 2]
```diff
@@ -90,6 +90,11 @@ interface mixin GlobalEventHandlers {
            attribute EventHandler onwaiting;
 };
 
+// https://drafts.csswg.org/css-animations/#interface-globaleventhandlers-idl
+partial interface mixin GlobalEventHandlers {
+           attribute EventHandler onanimationend;
```
[the remark this answers is not in the document]
Comment (mrobinson) [reply components/script/dom/webidls/EventHandler.webidl:95]: Yeah, the indentation in some of the webidls files is a bit bizarre. I think it's due to copy-and-paste effects from specification documents. :smile:  I've tried to match the indentation of the lines surrounding this one.

File: components/script/script_thread.rs (hunk 1 above)
Comment (mrobinson) [reply components/script/script_thread.rs:2929]: Oof. Thanks for pointing this out. :sweat_smile:

**Pull request merged**